simulator/property_collector_test.go

```diff
@@ -846,6 +847,53 @@ func TestPropertyCollectorInvalidSpecName(t *testing.T) {
 	}
 }
 
+func TestPropertyCollectorInvalidProperty(t *testing.T) {
+	tests := []struct {
+		name          string
+		path          string
+		expectedFault types.BaseMethodFault
+	}{
+		{
+			"specify property of array property",
+			"config.hardware.device.key",
+			new(types.InvalidProperty),
+		},
+	}
+
+	for _, test := range tests {
+		test := test // assign to local var since loop var is reused
+
+		t.Run(test.name, func(t *testing.T) {
+			m := ESX()
+
+			Test(func(ctx context.Context, c *vim25.Client) {
+				pc := property.DefaultCollector(c)
+
+				vm := Map.Any("VirtualMachine")
+				vmRef := vm.Reference()
+
+				var vmMo mo.VirtualMachine
+				err := pc.RetrieveOne(ctx, vmRef, []string{test.path}, &vmMo)
+				if err == nil {
+					t.Fatal("expected error")
+				}
+
+				// NOTE: since soap.vimFaultError is not exported, use interface literal for assertion instead
```

Comment:
You can use `soap.IsVimFault` and `soap.IsVimFault`, but ok with me to leave this as-is for this test: https://github.com/vmware/govmomi/blob/04d8df4b5f90b7826796baef7aed7ca81138170d/vim25/soap/error.go#L122-L129